phylogenetic/defaults/clade-i/config.yaml

```diff
@@ -23,10 +23,8 @@ filter:
 
 ### Filter to only Clade I sequences
 subsample:
-  everything:
-    group_by: ""
-    sequences_per_group: ""
-    query: "'clade in [\"I\", \"Ia\", \"Ib\"]'"
+  everything: >-
+    --query 'clade in ["I", "Ia", "Ib"]'
```

Comment:
> I think the most controversial change will be https://github.com/nextstrain/mpox/commit/f1ed6e987adea66e275dabf4ed9c1278a6561065, which switches to the generic subsample config schema. Seeking feedback on whether this is the correct direction.

I think the changes here are good - as you note, they align with [the linked docs](https://docs.nextstrain.org/en/latest/guides/bioinformatics/filtering-and-subsampling.html#generalizing-subsampling-in-a-workflow).  

These changes are backwards incompatible, so I'd suggest adding a check to the `config.smk` (see other comment thread) which enforces that `subsample.<sample>` must be a string value (and points to a piece of documentation explaining what to change if it's not).

Reply:
Sounds good. I'll consolidate all the config munging and add some config validation.

Reply:
It'd be possible to make these changes backwards compatible by adding a function to convert the old subsampling config syntax to the new one -- the code's already there to do this, it just needs a little rearranging. I'll leave it up to you to decide how important it is keep things backwards compatible for this repo.

Reply:
I decided to keep things backwards compatible and centralized everything config.smk in https://github.com/nextstrain/mpox/pull/321/commits/06af354c107c593c4e7027f5ab57f04cb6246635

Reply:
Haven't tested the code, but the approach looks great and reads nicely.